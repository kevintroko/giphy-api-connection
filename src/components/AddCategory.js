import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({setCategories}) => {
    const [value, setValue] = useState('');
    
    const handleInputChange = (e) => {
        console.log(e.target.value);
        setValue(e.target.value)
    }
    
    const handleSubmit = (e) => {
        e.preventDefault();
        
        if (value.trim().length > 2) {
            setCategories((categories) => [...categories, value]);
            setValue('');
        }
    }
    
    return (
    <form onSubmit={handleSubmit}>        
        <input 
            type="text"
            value={value}
            onChange={ handleInputChange }
        />
    </form>
)};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
