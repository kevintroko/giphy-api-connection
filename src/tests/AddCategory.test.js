import React from 'react';
import { shallow } from "enzyme";
import AddCategory from '../components/AddCategory';

describe('Fist app', () => {
        let wrapper;
    
        beforeEach(() => {    
            wrapper = shallow(<AddCategory setCategories={() => {}}/>); 
        });
        
        test('should match snapshot', () => {  
            expect(wrapper).toMatchSnapshot();
        });    
});