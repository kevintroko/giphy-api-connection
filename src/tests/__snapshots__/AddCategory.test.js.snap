// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fist app should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
